src/panels/lovelace/common/validate-condition.ts

```diff
@@ -58,32 +160,56 @@ function checkStateCondition(
     condition.entity && hass.states[condition.entity]
       ? hass.states[condition.entity].state
       : UNAVAILABLE;
+  let value = condition.state ?? condition.state_not;
+
+  // Handle entity_id, UI should be updated for conditionnal card (filters does not have UI for now)
+  if (Array.isArray(value)) {
+    value = value.map((v) => {
+      if (typeof v === "string" && isValidEntityId(v) && hass.states[v]) {
+        v = hass.states[v]?.state;
+      }
+      return `${v}`;
+    });
+  } else if (
+    typeof value === "string" &&
+    isValidEntityId(value) &&
+    hass.states[value]
+  ) {
+    value = hass.states[value]?.state;
+  }
 
   return condition.state != null
-    ? ensureArray(condition.state).includes(state)
-    : !ensureArray(condition.state_not).includes(state);
+    ? ensureArray(value).includes(state)
+    : !ensureArray(value).includes(state);
 }
 
 function checkStateNumericCondition(
   condition: NumericStateCondition,
   hass: HomeAssistant
 ) {
-  const entity =
-    (condition.entity ? hass.states[condition.entity] : undefined) ?? undefined;
+  const state = (condition.entity ? hass.states[condition.entity] : undefined)
+    ?.state;
+  let value = condition.above ?? condition.below;
 
-  if (!entity) {
-    return false;
+  // Handle entity_id, UI should be updated for conditionnal card (filters does not have UI for now)
+  if (
+    typeof value === "string" &&
+    isValidEntityId(value) &&
+    hass.states[value]
+  ) {
+    value = hass.states[value]?.state;
   }
 
-  const numericState = Number(entity.state);
+  const numericState = Number(state);
+  const numericValue = Number(value);
 
-  if (isNaN(numericState)) {
+  if (isNaN(numericState) || isNaN(numericValue)) {
     return false;
   }
 
   return (
-    (condition.above == null || condition.above < numericState) &&
-    (condition.below == null || condition.below > numericState)
+    (condition.above && numericValue < numericState) ||
+    (condition.below && numericValue > numericState)
```

Comment:
Add back `null`/`undefined` check to allow `0`

Reply:
If above and below are defined, we should check if the number is between. Why was this condition changed?

Reply:
fixed in f2f9ddce337cdb0d796b7ea0fdb0d37d21a9a476)